### ISSUE TITLE
aws_cognito_user_pool should export a provider name

### ISSUE THREAD

**Issue by youngbob** [terraform-providers/terraform-provider-aws#4700]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Cognito User Pools have a provider/issuer/endpoint name (I can't find a consistent name for this) of the format `cognito-idp.<region>.amazonaws.com/<YOUR_USER_POOL_ID>`. This name is used by various clients (as the token issuer, and to form the URL to retrieve the JWKs), and for configuring the user pool as an identity provider in a Cognito Identity Pool.

### New or Affected Resource(s)

* aws_cognito_user_pool

### Potential Terraform Configuration

```hcl
resource "aws_cognito_user_pool" "pool" {
  name = "pool"
}

resource "aws_cognito_user_pool_client" "client" {
  name  = "client"

  user_pool_id = "${aws_cognito_user_pool.pool.id}"
}

resource "aws_cognito_identity_pool" "main" {
  identity_pool_name  = "identity pool"

  cognito_identity_providers {
    client_id     = "${aws_cognito_user_pool_client.client.id}"
    provider_name = "${aws_cognito_user_pool.pool.endpoint}"
  }
}

output "user_pool_issuer" {
  value = "https://${aws_cognito_user_pool.pool.endpoint}"
}

output "user_pool_jwks" {
  value = "https://${aws_cognito_user_pool.pool.endpoint}/.well-known/jwks.json"
}
```

### References

* https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-integrating-user-pools-with-identity-pools.html#amazon-cognito-integrating-user-pools-with-identity-pools-using
* https://docs.aws.amazon.com/cognito/latest/developerguide/amazon-cognito-user-pools-using-tokens-with-identity-providers.html#amazon-cognito-identity-user-pools-using-id-and-access-tokens-in-web-api

**Comment by bflad** [comment]
The new `endpoint` attribute will be available in v1.22.0 of the AWS provider, releasing middle of this week.

**Comment by bflad** [comment]
This has been released in [version 1.22.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1220-june-05-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
